am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -142,76 +144,66 @@ public void revokeResourceAccess(@NotNull @Valid ExplicitAccessMetadata explicit
-                                                 @NotEmpty Set<@NotBlank String> userRoles,
-                                                 @NotNull @Valid Resource resource) {
-        if (userRoles.size() > 1) {
-            throw new IllegalArgumentException("Currently a single role only is supported. "
-                + "Future implementations will allow for multiple roles.");
-        }
-
+    public FilteredResourceEnvelope filterResource(@NotBlank String userId,
+                                                   @NotEmpty Set<@NotBlank String> userRoles,
+                                                   @NotNull @Valid Resource resource) {
         List<ExplicitAccessRecord> explicitAccess = jdbi.withExtension(AccessManagementRepository.class,
             dao -> dao.getExplicitAccess(userId, resource.getResourceId()));
 
         Map<JsonPointer, Set<Permission>> attributePermissions;
+        AccessType accessManagementType;
 
         if (explicitAccess.isEmpty()) {
-            List<RoleBasedAccessRecord> roleBasedAccess = jdbi.withExtension(AccessManagementRepository.class,
-                dao -> dao.getRolePermissions(
-                    resource.getType().getServiceName(),
-                    resource.getType().getResourceType(),
-                    resource.getType().getResourceName(),
-                    userRoles.iterator().next()));
-
-            if (roleBasedAccess.isEmpty()) {
-                return null;
-            }
+            attributePermissions = getRolePermissions(resource.getType(), userRoles);
 
-            if (explicitAccessType(userRoles)) {
+            if (attributePermissions == null) {
                 return null;
             }
 
-            attributePermissions = roleBasedAccess.stream().collect(getMapCollector());
+            accessManagementType = AccessType.ROLE_BASED;
 
         } else {
             attributePermissions = explicitAccess.stream().collect(getMapCollector());
+            accessManagementType = AccessType.EXPLICIT;
         }
 
         JsonNode filteredJson = filterService.filterJson(resource.getResourceJson(), attributePermissions);
 
-        return FilterResourceResponse.builder()
+        return FilteredResourceEnvelope.builder()
             .resourceId(resource.getResourceId())
+            .resourceDefinition(resource.getType())
             .data(filteredJson)
-            .permissions(attributePermissions)
+            .access(AccessEnvelope.builder()
+                .permissions(attributePermissions)
+                .accessManagementType(accessManagementType)
+                .build())
             .build();
     }
 
-    private boolean explicitAccessType(Set<String> userRoles) {
-        return jdbi.withExtension(AccessManagementRepository.class,
-            dao -> dao.getRoleAccessType(userRoles.iterator().next())).equals(AccessType.EXPLICIT);
+    private boolean roleBasedAccessType(String userRole) {
```

Comment:
why not `isRoleBasedAccess`?

Reply:
We should optimise it a bit I suppose. At the moment for 5 roles we make 5 calls to DB which might not be optimal. Simply number of DB requests depends on number of roles. Making one call with set of roles that filters out explicit roles might be more efficient. Ideally if microbenchmark confirmed it.

Reply:
Method should be moved under `getRolePermissions` to make reading easier as now `getRolePermissions` method is the only caller.